### PR TITLE
Add back setup.py to enable custom dynamic builds

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: .github/workflows/build-and-release-single-package.yml [hunk 1]
```diff
@@ -178,10 +167,3 @@ jobs:
         shell: bash
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Ensure Taipy release is marked as latest
-        run: |
-           gh release edit ${{needs.fetch-versions.outputs.taipy_VERSION}} --latest
-        shell: bash
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jrobinAV) [inline .github/workflows/build-and-release-single-package.yml]: I don't get why this is removed.

File: pyproject.toml [hunk 2]
```diff
@@ -4,14 +4,13 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy"
-version = "0.0.0"  # will be dynamically set
 description = "A 360° open-source platform from Python pilots to production-ready web apps."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline pyproject.toml]: We want to keep the `package_desc.md` here.

We differentiate the GitHub README from the package description file (PyPi description) used as project metadata. The README is used as GitHub's landing page and includes some marketing considerations not related to the Taipy package.

File: .github/workflows/build-and-release.yml [hunk 3]
```diff
@@ -249,11 +242,6 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - uses: stefanzweifel/git-auto-commit-action@v5
-        with:
-          file_pattern: '*/version.json'
-          commit_message: Update version to ${{ needs.fetch-versions.outputs.NEW_VERSION }}
-
```
Comment (jrobinAV) [inline .github/workflows/build-and-release.yml]: Why don't we want to keep that?

File: taipy/config/pyproject.toml [hunk 4]
```diff
@@ -4,14 +4,13 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy-config"
-version = "0.0.0"  # will be dynamically set
 description = "A Taipy package dedicated to easily configure a Taipy application."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline taipy/config/pyproject.toml]:
```suggestion
readme = "package_desc.md"
```

File: taipy/core/pyproject.toml [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-requires = ["setuptools>=42", "wheel", ]
+requires = ["setuptools>=42", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy-core"
-version = "0.0.0"   # will be dynamically set
 description = "A Python library to build powerful and customized data-driven back-end applications."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline taipy/core/pyproject.toml]:
```suggestion
readme = "package_desc.md"
```

File: taipy/gui/pyproject.toml [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
-requires = [ "setuptools>=42", "wheel", "setuptools_scm",]
+requires = ["setuptools>=42", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy-gui"
-version = "0.0.0"  # will be set dynamically
 description = "Low-code library to create graphical user interfaces on the Web for your Python applications."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline taipy/gui/pyproject.toml]:
```suggestion
readme = "package_desc.md"
```

File: taipy/rest/pyproject.toml [hunk 7]
```diff
@@ -4,14 +4,13 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy-rest"
-version = "0.0.0"  # will be dynamically set
 description = "Library to expose taipy-core REST APIs."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline taipy/rest/pyproject.toml]:
```suggestion
readme = "package_desc.md"
```

File: taipy/templates/pyproject.toml [hunk 8]
```diff
@@ -4,14 +4,13 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "taipy-templates"
-version = "0.0.0"  # will be dynamically set
 description = "An open-source package holding Taipy application templates."
-readme = "package_desc.md"
+readme = "README.md"
```
Comment (jrobinAV) [inline taipy/templates/pyproject.toml]:
```suggestion
readme = "package_desc.md"
```

File: .github/workflows/build-and-release-single-package.yml (hunk 1 above)
Comment (joaoandre-avaiga) [reply .github/workflows/build-and-release-single-package.yml]: Oh thats a mistake. Im fixing it

**Review by jrobinAV — DISMISSED** [review]
LGTM. Thx

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**